File: 1.2.py
# La tabla del número X desde 1 hasta el 10

def solicitarNumero():
    numero = 0
    while numero == 0:
        try:
            numero = int(input("Ingresa un numero: "))
            return numero
        except ValueError:
            print("El valor ingresado no es un numero")

def tablasDeMultiplicar(numero):
    for i in range(1, 11):
        print(numero * i)

def main():
    numero = solicitarNumero()
    tablasDeMultiplicar(numero)

if __name__ == "__main__":
    main()

File: 1.2.kt
/* La tabla del número X desde 1 hasta el 10 */

fun pedirNumero(): Int {
    var numero: Int? = null

    while (numero == null) {
        print("Ingresa un número: ")
        numero = readLine()?.toIntOrNull()

        if (numero == null) {
            println("Debe ser un número entero válido.")
        }
    }

    return numero
}

fun tablasDeMultiplicar(numero: Int) {
    for (i in 1..10){
        val multiplicacion = numero * i
        println("$numero * $i = $multiplicacion")
    }
}

fun main() {
    val numero = pedirNumero()
    tablasDeMultiplicar(numero)
}

